content/docs/user-guide/assets/asset-processor/debugging.md

```diff
@@ -6,9 +6,256 @@ weight: 500
+
+To help identify the block of logic in the Product Asset that is changing each time it is processed, you may be able to use a standard file diffing tool using two generated product assets. You can quickly re-generate the product asset by right clicking it in the Asset Processor's asset tab and selecting reprocess source asset.
+
+Another debugging option is setting up automation to catch this early. If you have an automated system that processes assets for your project, you can expand this automation to also 
+
+### Same Product Asset generated with different Sub ID
+<a name="ProductAssetSubId"></a>
+
+#### Situation
+The Sub ID for a Product Asset changes when it is not expected to change.
+
+In the worst case, the Sub ID is changing when no changes have been made to the Source Asset or builder that created the Product Asset. In other cases, a change to the Source Asset results in a Product Asset that has a different Sub ID than it did previously.
+
+#### Effect
+Assets are mostly referenced by asset ID, the UUID of the source asset with the Sub ID for that product asset. If this is not consistent and the Sub ID changes for a Product Asset when it should not change, then external references to this asset can end up incorrect, pointing at a different asset after a rebuild or pointing at no asset if no other products use that Sub ID.
+
+#### Example
+Here is an example with scene files and prefabs:
+1. Create a new FBX file named shapes_1.fbx with two meshes in it, a cube and a cylinder. Name the cube Mesh_A and the cylinder Mesh_B.
+1. Swap the names of these two meshes, so the cube is now named Mesh_B and the cylinder is named Mesh_A. Save this to a second FBX named shapes_2.fbx.
+1. Place shapes_1.fbx in your project, and rename it to shapes.fbx.
+1. Create a new prefab with an entity in it with a mesh component.
+1. Assign the cube Mesh_A from shapes.fbx to this component.
+1. Save and quit the Editor.
+1. Replace the shapes.fbx file in your project with shapes_2.fbx.
+1. Load the Editor.
+1. Load the prefab you saved earlier.
+1. Notice that the mesh on the entiyt component is still Mesh_A, but now visually looks like the cylinder.
+
+This happens because scene processing tracks the meshes by name, and by swapping the names between the meshes you've replaced which product assets are which. The prefab tracks the outgoing product asset reference via asset ID. When I tested this locally, my Mesh_A azmodel had the Product Asset ID {00431C65-528C-570B-8A52-5D43679BBFAC}:270123712, and Mesh_B has the asset ID {00431C65-528C-570B-8A52-5D43679BBFAC}:285065766. When I swapped the FBX files, the node within the file named Mesh_A still had that same first ID, but the geometry was a cylinder now, changing what it looked like in the prefab.
```

Comment:
The end of the first sentence is confusing.  I think a comma after "between the meshes" might help (assuming I've understood correctly) but a slight rewording might make it more clear.

Reply:
I've reworded this